lib/wda/webdriveragent.js

```diff
@@ -74,6 +75,45 @@ class WebDriverAgent {
     log.info(`Using WDA agent: '${this.agentPath}'`);
   }
 
+  async cleanupObsoleteProcesses () {
+    const pids = await getPIDsListeningOnPort(this.url.port,
+      (cmdLine) => (cmdLine.includes('/WebDriverAgentRunner') || cmdLine.includes('/iproxy')) &&
+        !cmdLine.toLowerCase().includes(this.device.udid.toLowerCase()));
+    if (!pids.length) {
+      log.debug(`No obsolete cached processes from previous WDA sessions ` +
+                `listening on port ${this.url.port} have been found`);
+      return;
+    }
+
+    log.info(`Detected ${pids.length} obsolete cached process${pids.length === 1 ? '' : 'es'} ` +
+             `from previous WDA sessions. Cleaning up...`);
+    try {
+      await exec('kill', pids);
```

Comment:
https://www.npmjs.com/package/fkill is a really good package for killing processes.

Reply:
is it already used somewhere else in this module?

Reply:
replaced exec with fkill